packages/flutter_tools/test/general.shard/config_test.dart

```diff
@@ -81,4 +81,29 @@ void main() {
     expect(file.existsSync(), false);
     expect(bufferLogger.errorText, contains('Failed to decode preferences'));
   });
+
+
+  testWithoutContext('Config in home dir', () {
+    memoryFileSystem.file('.flutter_example').writeAsStringSync('{"hello":"bar"}');
+    config = Config(
+      'example',
+      fileSystem: memoryFileSystem,
+      logger: BufferLogger.test(),
+      platform: fakePlatform,
+    );
+    expect(config.getValue('hello'), 'bar');
+    expect(memoryFileSystem.file('.config/flutter/example').existsSync(), false);
+  });
+
+  testWithoutContext('Config in config dir', () {
```

Comment:
Longer comment here: config is created in the specified xdg directory if it does not already exist in the home dir?